### PR TITLE
Add possibility of setting the path to Sublime Merge

### PR DESCRIPTION
**Pull request by Zwyx**
Addresses https://github.com/adhamu/history-in-sublime-merge/issues/15

This adds the extension setting `history-in-sublime-merge.path`, which allows to override the path to Sublime Merge:

image

### CODE REVIEW

File: src/extension.ts
Comment (adhamu) [inline src/extension.ts]: This can just be:

```typescript
const path = customPath || SMERGE_BINARY_PATH
```

File: README.md
Comment (adhamu) [inline README.md]: I think this can be removed. These requirements are accurate for other operating systems

File: src/extension.ts
Comment (adhamu) [reply src/extension.ts]: Or even just

```typescript
child.execFile(customPath || SMERGE_BINARY_PATH, args, {
  cwd: repository,
})
```

**Review by adhamu — DISMISSED** [review]

File: src/extension.ts
Comment (Zwyx) [reply src/extension.ts]: Nice but `customPath` is of type `unknow` though, making the check necessary:

image

File: README.md
Comment (Zwyx) [reply README.md]: No worries, done :+1:

File: src/extension.ts
Comment (adhamu) [reply src/extension.ts]: `get` accepts a generic so you could do

```typescript
const customPath = vscode.workspace
     .getConfiguration()
     .get<string>('history-in-sublime-merge.path');
```

If you hover over `customPath`, it should be typed as `string | undefined`

Comment (Zwyx) [reply src/extension.ts]: Nice, done :+1:

**Review by adhamu — APPROVED** [review]

**Comment by adhamu** [comment]
Feel free to merge this when you're ready and I'll get a new version published on the marketplace as soon as I can (I don't have CI/CD setup for this)

**Comment by Zwyx** [comment]
Thanks! I'm not able to merge though, GitHub says that _Only those with write access to this repository can merge pull requests._

I'm ready, you can merge when you want!

**Comment by adhamu** [comment]
Version 1.2.18 has been published to the marketplace
https://marketplace.visualstudio.com/items?itemName=adhamu.history-in-sublime-merge

**Comment by Zwyx** [comment]
Awesome! Thanks for having taken care of it so quickly, really appreciated!

**Comment by adhamu** [comment]
> Awesome! Thanks for having taken care of it so quickly, really appreciated!

No worries. Thanks for the contribution

**Pull request merged**
